test/map.test.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+var test = require('tape');
+
+// run tests on builtin Map as well as native C++ implementation
+[['builtin', Map], ['native', require('../index.js')]].forEach(([mapType, Map]) => {
+
+  test(`test ${mapType} constructor`, (assert) => {
+    assert.doesNotThrow(() => {new Map()}, 'can construct an empty new Map()')
+    // see array constructor tests below
+    assert.throws(() => {new Map([1,2,3,4]);}, TypeError, 'cannot construct a Map from a flat array');
+    assert.throws(() => {new Map({1:2, 3:4});}, TypeError, 'cannot construct a Map from an object');
+    assert.end();
+  });
+
+  test(`test ${mapType} set method`, (assert) => {
+    let myMap = new Map();
+    const startSize = myMap.size;
+
+    assert.doesNotThrow(() => {myMap.set('a string', "value")}, 'can set string key to a value');
+    assert.equal(myMap.size, startSize+1, 'setting key-value increases map size by 1');
+
+    assert.doesNotThrow(() => {myMap.set({}, 'value')}, 'can set object key to a value');
+    assert.doesNotThrow(() => {myMap.set(()=>{}, 'value')}, 'can set function key to a value');
+    assert.doesNotThrow(() => {myMap.set(null, 'value')}, 'can set null key to a value');
+
+    //
+    assert.doesNotThrow(() => {myMap.set(1)}, 'can call set with only 1 argument');
+    assert.ok(myMap.has(1) && myMap.get(1) === undefined, 'set with only 1 argument leaves value as undefined');
+    assert.doesNotThrow(() => {myMap.set(2,3,4)}, 'can call set with more than 2 arguments');
+    assert.ok(myMap.has(2) && myMap.get(2) === 3, 'set ignores arguments after 2nd');
+
+    assert.end();
+  });
+
+  test(`test ${mapType} get method`, (assert) => {
+    let m = new Map([[1, 2], [3, 4], [{}, 6]]);
+    assert.equal(m.get(1), 2, 'get returns the value associated with an existing key');
+    assert.equal(m.get(5), undefined, 'get returns undefined for a nonexistent key');
+    assert.equal(m.get({}), undefined, 'get returns undefined for nonidentical keys');
+    assert.end();
+  });
+
+  test('test has method', (assert) => {
+    let empty = new Map();
+    assert.notOk(empty.has('anything') || empty.has(1) || empty.has(''), "empty map has no keys");
+    let m = new Map([[1, 2], [3, 4], [()=>{}, 6], [{}, 8]]);
```

Comment:
It's great you're testing with a function and object here as keys. Would it be possible to show retrieval of these is possible (with a stable function/object reference for retrieval?)